File: 0881-boats-to-save-people/0881-boats-to-save-people.java
// idea is to try to fit heavy one first, and then fit lightest one with that person if possible
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int boats=0, left=0, right=people.length-1;

        while(left <= right){
            boats++;
            if(people[left] + people[right] <= limit) left++;
            right--;
        }
        
        return boats;
    }
}

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
# idea is to try to fit heavy one first, and then fit lightest one with that person if possible
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        boats, left, right = 0, 0, len(people)-1

        while left <= right:
            boats += 1
            if people[right] + people[left] <= limit: left += 1
            right -= 1
        return boats
